docs/dev/test-architecture.adoc

```diff
@@ -184,7 +184,13 @@ Context("when running help for storage command", func() {
 ----
 +
 
-* Due to parallel test run support make sure that the should run in isolation, otherwise the test result will lead to race condition. To achieve this ginkgo provides some in build functions `BeforeEach`, `AfterEach` etc.
+* Don't create a new test spec for the steps which can be run with the existing specs. Also try to avoid adding the functionality which are part of another feature test file, only add if it is a key requirement.
++ 
+
+* Spec level conditions, pre and post requirements should be run in ginkgo built-in tear down steps `JustBeforeEach` and `JustAfterEach`
++
+
+* Due to parallel test run support make sure that the spec should run in isolation, otherwise the test result will lead to race condition. To achieve this ginkgo provides some in build functions `BeforeEach`, `AfterEach` etc.
```

Comment:
Link to Ginkgo documentation. Don't use `etc.`

You can shorten this to:

* Some tests are unable to be run in parallel. You can achieve this by running XX (link to Ginkgo doc or example) with your test.

Or something like that.

Reply:
> Some tests are unable to be run in parallel.

But I would say stating out the reason best explain the use of `BeforeEach`, `AfterEach` and so on. For adding link I would again say the same as in https://github.com/openshift/odo/pull/3726#discussion_r468378476

Reply:
You can combine ```* Spec level conditions, pre and post requirements should be run in ginkgo built-in tear down steps `JustBeforeEach` and `JustAfterEach```` and ```* Due to parallel test run support make sure that the spec should run in isolation, otherwise the test result...``` like 

You can combine ```* Spec level conditions, pre and post requirements should be run in ginkgo built-in tear down steps `JustBeforeEach` and `JustAfterEach. To make spec isolation for parallel run you can use and update the same teardown steps as per the requirement.